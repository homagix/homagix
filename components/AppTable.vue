<script setup lang="ts">
defineProps<{
  list: unknown[]
  columnNames: string[]
}>()
</script>

<template>
  <table>
    <thead>
      <tr>
        <th v-for="name in columnNames">{{ name }}</th>
      </tr>
    </thead>
    <tbody>
      <tr v-for="entry in list">
        <slot :entry="entry" />
      </tr>
    </tbody>
  </table>
</template>

<style lang="scss" scoped>
table {
  border: 1px solid lightgrey;
  border-collapse: collapse;

  thead th {
    text-align: left;
  }

  tbody tr:nth-child(odd) {
    background-color: #fee0b9;
  }

  :deep(td), th {
    padding: 3px 6px;
  }
}
</style>
