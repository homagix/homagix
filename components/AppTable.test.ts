import { describe, it, expect } from "vitest"
import { setup } from "@nuxt/test-utils"
import { mountSuspended } from "@nuxt/test-utils/runtime"
import AppTable from "./AppTable.vue"

type Entry = { firstName: string; repo: string }

describe("AppTable", () => {
  setup({ server: false })

  it("renders as expected", async () => {
    const wrapper = await mountSuspended(AppTable, {
      props: {
        columnNames: ["First name", "Repository"],
        list: [
          { firstName: "Arnie", repo: "https://github.com/arnie/recipes" },
          { firstName: "John", repo: "https://github.com/john/recipes" },
        ],
      },
      slots: {
        default: ({ entry }) => JSON.stringify(entry),
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
