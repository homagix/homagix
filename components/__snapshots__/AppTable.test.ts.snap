// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppTable > renders as expected 1`] = `
"<table data-v-95accda9="">
  <thead data-v-95accda9="">
    <tr data-v-95accda9="">
      <th data-v-95accda9="">First name</th>
      <th data-v-95accda9="">Repository</th>
    </tr>
  </thead>
  <tbody data-v-95accda9="">
    <tr data-v-95accda9="">{"firstName":"Arnie","repo":"https://github.com/arnie/recipes"}</tr>
    <tr data-v-95accda9="">{"firstName":"John","repo":"https://github.com/john/recipes"}</tr>
  </tbody>
</table>"
`;
